std/algorithm/searching.d

```diff
@@ -122,7 +122,7 @@ template all(alias pred = "a")
     if (isInputRange!Range)
     {
         static assert(is(typeof(unaryFun!pred(range.front))),
-                "`" ~ pred.stringof[1..$-1] ~ "` isn't a unary predicate function for range.front");
+                "`" ~ pred.stringof ~ "` isn't a unary predicate function for range.front");
```

Comment:
String predicates are bad practice and will most likely be deprecated at some point, so from that point of view this change is fine. However, you could check whether the predicate [isSomeString](https://phobos.dpldocs.info/std.traits.isSomeString.html) and eliminate the quotes based on that. Something along the lines of  ` isSomeString!pred ? pred.stringof : pred.stringof[1..$-1]`. That way we support both cases graciously.

Reply:
Done. My only reluctance to go this route was that it'd make different cases look the same, but realistically string predicates will still look distinct in error messages due to the 'a' metavariable.

Updated drepl transcript:
```d
D> import std;
std
D> "hi".all!isWhite
static assert:  "`isWhite` isn't a unary predicate function for range.front"
/tmp/drepl.mzCScR/_mod1.d(10):        instantiated from here: `all!string`

D> "hi".all!(c => c.isWhite)
static assert:  "`__lambda2` isn't a unary predicate function for range.front"
/tmp/drepl.mzCScR/_mod1.d(10):        instantiated from here: `all!string`

D> "hi".all!"a.isWhite"
static assert:  "`a.isWhite` isn't a unary predicate function for range.front"
/tmp/drepl.mzCScR/_mod1.d(10):        instantiated from here: `all!string`
```